app/javascript/utilities/gist.js

```diff
@@ -0,0 +1,49 @@
+export function embedGists() {
+  const waitingOnPostscribe = setInterval(() => {
+    clearInterval(waitingOnPostscribe);
+    const gistTags = document.querySelectorAll('.ltag_gist-liquid-tag');
+
+    // Only load scripts for gists if needed
+    if (gistTags.length > 0) {
+      import('postscribe').then(({ default: postscribe }) => {
+        for (const gistTag of gistTags) {
+          postscribe(gistTag, gistTag.firstElementChild.outerHTML, {
+            beforeWrite: function (context) {
+              return function (text) {
+                if (context.childElementCount > 3) {
+                  return "";
+                }
+                return text;
+              }
+            }(gistTag)
+          });
+        }
+      });
+    }
+  }, 500);
+}
+
+export function embedGistsInComments() {
+  // allows for getting the gist embed after new comment submit and preview
+  const commentForm = document.getElementById("new_comment");
+  const previewBtn = document.getElementsByClassName('preview-toggle')[0];
+  const dismissBtn = document.querySelector('.dismiss-comment');
+
+  if (commentForm && typeof commentForm !== "undefined") {
+    commentForm.addEventListener("submit", (_event) => {
+      embedGists();
+    });
+  }
+
+  if (previewBtn && typeof previewBtn !== "undefined") {
+    previewBtn.addEventListener("click", (_event) => {
+      embedGists();
+    });
+  }
+
+  if (dismissBtn && typeof dismissBtn !== "undefined") {
+    dismissBtn.addEventListener("click", (_event) => {
+      embedGists();
+    });
+  }
```

Comment:
Refactor (blocking): Use [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining). It's a newer feature of the language that simplifies the code by eliminating the defensive coding pattern. Well we're still doing defensive coding, but in a more concise way, via `?`.

```suggestion
  // allows for getting the gist embed after new comment submit
  document.getElementById("new_comment")?.addEventListener("submit", (_event) => {
    embedGists();
  });

  document.querySelector('.preview-toggle')?.addEventListener("click", (_event) => {
    embedGists();
  });
```

Reply:
@nickytonline thanks for your review, I've added those changes and a missing case.

Sometimes I get errors with Travis such as : 

```
  Failure/Error: expect(result).to be_empty

       expected `[{:author=>{:name=>"Ali Fritsch", :profile_image_90=>"/uploads/user/profile_image/2095/2eed66b4-b3c4-...:slug=>"no-country-for-old-men-772-3095", :tags=>[], :title=>"Awesome New Listing", :user_id=>2095}].empty?` to be truthy, got false
``` 

It seems unrelated to this PR and it appears randomly. Do you confirm that?

Reply:
It seems unrelated and locally the test passes. I'll restart the build